pkg/querier/distributor_queryable.go

```diff
@@ -148,18 +148,20 @@ func (q *distributorQuerier) Select(_ bool, sp *storage.SelectHints, matchers ..
 	}
 
 	// Using MatrixToSeriesSet (and in turn NewConcreteSeriesSet), sorts the series.
-	return series.MatrixToSeriesSet(matrix)
+	return series.MatrixToSeriesSet(true, matrix)
```

Comment:
Why this is always true?

Reply:
It should be fine to pass down the flag here as well as the method `queryIngesters` is also  making sure it does not return duplicated series.

https://github.com/cortexproject/cortex/blob/dd4240d6d02a48bdd732b57a266b0c4192bb5e1e/pkg/distributor/query.go#L185-L201

I just passed true there as this is a old code that should be removed in the future (at least on my opnion) as streaming chunks has a way better performance. 

But yah.. just for consistency lets pass the flag down here as well.